lib/js/app/modules/app/saga.ts

```diff
@@ -246,6 +249,32 @@ export function* appStart({ payload }: AppStartAction) {
   yield spawn(watchScreenResize);
 }
 
+export function* generateFileName() {
+  const savedQuery = yield select(getSavedQuery);
+  const query = yield select(getQuerySettings);
+
+  let fileName = 'chart';
+  if (savedQuery?.name) {
+    fileName = `${savedQuery.name}-${Date.now()}`;
+  } else if (query?.analysis_type && query?.event_collection) {
+    fileName = `${query.analysis_type}-${query.event_collection}-${Date.now()}`;
+  }
+  return fileName;
+}
+
+export function* exportChartToImage() {
+  const node = document.getElementById('query-visualization');
+  if (!node) throw new Error('Query visualization container is not available');
+  const fileName = yield generateFileName();
+  exportToImage({ fileName, node });
```

Comment:
As we here depend on external library I suggest to wrap `exportToImage` with try catch block and handle exception with toast notification that informs user about failure export operation.